mmdet3d/models/roi_heads/bbox_heads/h3d_head.py

```diff
@@ -0,0 +1,1017 @@
+            mean_size_arr=bbox_coder['mean_sizes'],
+            num_proposal=num_proposal,
+            with_angle=bbox_coder['with_rot'],
+            **proposal_module_cfg)
+
+    def init_weights(self, pretrained=None):
+        """Initialize the weights in detector.
+
+        Args:
+            pretrained (str, optional): Path to pre-trained weights.
+                Defaults to None.
+        """
+        pass
+
+    def forward(self, feat_dict, sample_mod):
+        """Forward pass.
+
+        Args:
+            feat_dict (dict): Feature dict from backbone.
+            sample_mod (str): Sample mode for vote aggregation layer.
+                valid modes are "vote", "seed" and "random".
+
+        Returns:
+            dict: Predictions of vote head.
+        """
+        aggregated_points = feat_dict['aggregated_points']
+        refine_predictions, refine_dict = self.refine_module(feat_dict)
+        refine_decode_res = self.bbox_coder.split_pred(refine_predictions,
+                                                       aggregated_points)
+        for key in refine_decode_res.keys():
+            refine_dict[key + '_opt'] = refine_decode_res[key]
+        return refine_dict
+
+    def loss(self,
+             bbox_preds,
+             points,
+             gt_bboxes_3d,
+             gt_labels_3d,
+             pts_semantic_mask=None,
+             pts_instance_mask=None,
+             img_metas=None,
+             rpn_targets=None,
+             gt_bboxes_ignore=None):
+        """Compute loss.
+
+        Args:
+            bbox_preds (dict): Predictions from forward of vote head.
+            points (list[torch.Tensor]): Input points.
+            gt_bboxes_3d (list[:obj:`BaseInstance3DBoxes`]): Ground truth \
+                bboxes of each sample.
+            gt_labels_3d (list[torch.Tensor]): Labels of each sample.
+            pts_semantic_mask (None | list[torch.Tensor]): Point-wise
+                semantic mask.
+            pts_instance_mask (None | list[torch.Tensor]): Point-wise
+                instance mask.
+            img_metas (list[dict]): Contain pcd and img's meta info.
+            rpn_targets (Tuple) : Targets generated by rpn head.
+            gt_bboxes_ignore (None | list[torch.Tensor]): Specify
+                which bounding.
+
+        Returns:
+            dict: Losses of H3dnet.
+        """
+        (vote_targets, vote_target_masks, size_class_targets, size_res_targets,
+         dir_class_targets, dir_res_targets, center_targets, mask_targets,
+         valid_gt_masks, objectness_targets, objectness_weights,
+         box_loss_weights, valid_gt_weights) = rpn_targets
+
+        loss_inputs = (bbox_preds, points, gt_bboxes_3d, gt_labels_3d,
+                       pts_semantic_mask, pts_instance_mask, img_metas,
+                       gt_bboxes_ignore)
+        losses = {}
+
+        # calculate refined proposal loss
+        refined_proposal_loss = self.get_proposal_stage_loss(
```

Comment:
what is the different between `self.get_proposal_stage_loss` and loss in VoteHead

Reply:
support paramaters `suffix`, inorder to compute bbox regression loss and classification loss for different stages.

Reply:
And also remove the loss of `vote_module`